tfhe/src/js_on_wasm_api/js_high_level_api/mod.rs

```diff
@@ -2,6 +2,7 @@ use wasm_bindgen::prelude::*;
 
 pub(crate) mod config;
 pub(crate) mod integers;
+#[allow(clippy::use_self)]
```

Comment:
this can probably be fixed without too much of a hassle ?

Reply:
Added a comment to explain why not (using Self does not work well with #[wasm_bindgen] macro)

Reply:
I don't think its possible for all in the wasm part.

e.g:

```
#[wasm_bindgen]
impl TfheClientKey {
    #[wasm_bindgen]
    pub fn generate(config: &TfheConfig) -> Result<TfheClientKey, JsError> {
        catch_panic(|| Self(hlapi::ClientKey::generate(config.0.clone())))
    }
}
```

We would want to return `Result<Self, JsError>`, but `#[wasm_bindgen]` does not allow that so we have to keep `Result<TfheClientKey, JsError>`